### DISCUSSION TITLE
[Collation] Final Deliverables Checklist

### DISCUSSION THREAD

**Discussion by sp4ce-cowboy** [AY2324S1-CS2103-T16-3/tp#209] [General]
### User Guide due **14 Nov 1 PM**
- [x] Check for grammar/spelling errors
- [x] Compare with AB3 [here](https://se-education.org/addressbook-level3/UserGuide.html)
- [x] Need to highlight the find + edit = disappearing transaction issue in the edit transaction section
- [x] Need to update some old outputs based on the previous UniCa$h version (before multiple command words
were added)
- [x] Need to hyperlink wherever applicable for easier navigation
![image](https://github.com/AY2324S1-CS2103-T16-3/tp/assets/19762596/82cc8fc1-b8ad-4500-a9e2-ad6a4fcf147d)
- [ ] Need to edit some example command outputs that are too long to fit in the PDF format

![image](https://github.com/AY2324S1-CS2103-T16-3/tp/assets/19762596/cba8f418-0163-4ad3-a6cf-f8787b579aad)

### Developer Guide due **14 Nov 1 PM**
- [x] Check for grammar/spelling errors
- [x] Compare with AB3 [here](https://se-education.org/addressbook-level3/DeveloperGuide.html)
- [x] Need to add Appendix: Planned Enhancements (for countering feature flaws). See [this](https://github.com/AY2324S1-CS2103-T16-3/tp/pull/211#discussion_r1390645575) comment.
- [  ] [Optional] Can add Planned Features section 
- [x] Need to add Appendix: Efforts
- [ ] Need to add Instructions for Manual Testing.
- [x] Write user stories within the DG

![image](https://github.com/AY2324S1-CS2103-T16-3/tp/assets/19762596/1b87f850-ff62-4939-a839-6d3116ba16fc)

ALL DIAGRAMS -> Double check if screenshots are correct, and if the text size matches surrounding text
UniCash v UniCa$h

### Code Quality  **14 Nov 1 PM**
- Rmb to double check own code quality [Check for the below in own classes & features]
    - Logging, Assertions, Defensive Coding, SLAP
    - Naming of methods/variables & unit tests naming
    - Other code style -> duplicate methods, unused code, etc
- Unit Testing
    - Make sure all unit test names are ok
    - Make sure boundary cases are identified
- Javadocs
    - Javadocs wherever missing (for public methods)

### Other submissions

- PPP (must be 2 pages, use Google Chrome to export to PDF like [this](https://se-education.org/guides/tutorials/savingPdf.html))
- Demo video Wed 15 Nov 2359
- PE Readiness Quiz & Code Reuse Declaration

**Comment by sp4ce-cowboy** [comment]
### Planned Enhancements
- Expand find command to all properties
- Add confirmation to some data deletion commands (delete, clear, reset)
- Expand support for multiple transaction types
- Expand summary window to incomes as well
- Batch deletion/clear (by day, by month, etc)

**Reply by woojiahao** [reply]
<img width="959" alt="Screenshot 2023-11-13 at 12 55 32" src="https://github.com/AY2324S1-CS2103-T16-3/tp/assets/29497029/7a2cc9c3-237d-4211-8c67-c2c7583a038d">

Added the budget planned enhancements

**Reply by woojiahao** [reply]
All added under #211

**Comment by sp4ce-cowboy** [comment]
### [UG Bug]

![image](https://github.com/AY2324S1-CS2103-T16-3/tp/assets/19762596/6f63ca04-015a-42e1-a1ea-909033bdae78)

This part needs to be updated with the newer URL

**Reply by sp4ce-cowboy** [reply]
Nvm I've added it to #208

**Comment by woojiahao** [comment]
@sp4ce-cowboy Could you convert the bullet points to task items as such:

`- [ ] <text>`, this way, we can check off whatever we've done so we don't need to duplicate the work by accident

**Comment by sp4ce-cowboy** [comment]
### DG CHECKLIST (from course website)

These are considered DG bugs (if they hinder the reader):

Those given as possible UG bugs ...



 Architecture:

Symbols used are not intuitive
Indiscriminate use of double-headed arrows
architecture-level diagrams contain lower-level details
Description given are not sufficiently high-level


UML diagrams:

Notation incorrect or not compliant with the notation covered in the course.
Some other type of diagram used when a UML diagram would have worked just as well.
The diagram used is not suitable for the purpose it is used.
The diagram is too complicated.
 

Code snippets:

Excessive use of code e.g., a large chunk of code is cited when a smaller extract would have sufficed.


Problems in User Stories. Examples:

Incorrect format
All three parts are not present
The three parts do not match with each other
Important user stories missing


Problems in Use Cases. Examples:

Important use cases missing (a use case is important if it involves a user interaction that is worthy of documenting e.g., it has multiple extensions -- this is not the same as the feature being important)
Formatting/notational errors
Incorrect step numbering
Unnecessary UI details mentioned
Missing/unnecessary steps
Missing extensions


Problems in NFRs. Examples:

Not really a Non-Functional Requirement
Not scoped clearly (i.e., hard to decide when it has been met)
Not reasonably achievable
Highly relevant NFRs missing

Problems in Glossary. Examples:

Unnecessary terms included
Important terms missing

**Comment by sp4ce-cowboy** [comment]
@lipwei1808 @jamz903 @woojiahao @elhy1999 Closing this discussion, Good work everyone!
